setting.go

```diff
@@ -34,9 +34,8 @@ func newClient() *http.Client {
 
 // Client return the default underlying http client
 func (r *Req) Client() *http.Client {
-	if r.client == nil {
-		r.client = newClient()
-	}
+	r.mu.RLock()
+	defer r.mu.RUnlock()
 	return r.client
 }
```

Comment:
This would return a client not initialized.

``` bash
$ go test
--- FAIL: TestSetClient (0.00s)
    setting_test.go:57: transport is not http.Transport: <nil>
FAIL
exit status 1
FAIL    github.com/imroc/req    0.968s
```

Reply:
True , missed that will readd the initialization of r.client and with Lock instead of Rlock.